### PR TITLE
Update Python regression tests to match C# equivalents

### LINKED ISSUE #2192: Python regression algorithms differ from C# equivalents
#### Expected Behavior
Python regression algorithms should be identical to their C# equivalents.

#### Actual Behavior
8 Python regression algorithms have different parameters and/or code.

#### Potential Solution
Update the Python regression algorithms to match the C# equivalents.

#### Reproducing the Problem
Run the Python regression tests for:
- `BasicTemplateDailyAlgorithm`
- `BasicTemplateIntrinioEconomicData`
- `CoarseFineFundamentalRegressionAlgorithm`
- `CompositeAlphaModelFrameworkAlgorithm`
- `CustomDataRegressionAlgorithm`
- `DelistingEventsAlgorithm`
- `OptionSplitRegressionAlgorithm`
- `UpdateOrderRegressionAlgorithm`

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by StefanoRaggi**
#### Description
- Some Python regression tests have been updated to match their C# equivalents.
- A resolution bug was fixed in the Python version of `EmaCrossAlphaModel`.
- The regression order log file name now includes the algorithm name.
- The `PythonEngine.BeginAllowThreads()` method is now called only once per process launch, because it was causing a Python Fatal Error with multiple Python regression tests.
- A few `import` statements were fixed in a couple of Python algorithms.
- Added missing `try...catch` with logging in `AlgorithmRunner` task.
- Fixed `static` variable bug in `JobQueue`, causing Python regression algorithms to fail if ran after `AddRemoveOptionUniverseRegressionAlgorithm` (which does not have a Python version).

All regression tests (C# and Python) are finally passing, also when running the full suite.

#### Related Issue
Closes #2192 

#### Motivation and Context
8 Python regression tests were failing.

#### Requires Documentation Change
No.

#### How Has This Been Tested?
Regression test runs.

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/ect)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.Python/UpdateOrderRegressionAlgorithm.py [hunk 1]
```diff
@@ -117,18 +117,19 @@ def OnData(self, data):
...
+
         #fills update LastFillTime
         if (order.Status == OrderStatus.Filled or order.Status == OrderStatus.PartiallyFilled) and order.LastFillTime != orderEvent.UtcTime:
             raise ValueError("Expected filled order LastFillTime to equal fill order event time.")
 
-        #updates have a status of submitted and the created time will be different from the event time
-        if order.Status == OrderStatus.Submitted and order.Time != orderEvent.UtcTime and order.LastUpdateTime != orderEvent.UtcTime:
-            raise ValueError("Expected updated order LastUpdateTime to equal submitted update order event time")
+        # check the ticket to see if the update was successfully processed
+        if len([ur for ur in ticket.UpdateRequests if ur.Response is not None and ur.Response.IsSuccess]) > 0 and order.CreatedTime != self.UtcTime and order.LastUpdateTime == None:
```
Comment (jingwu74) [inline Algorithm.Python/UpdateOrderRegressionAlgorithm.py]: `order.LastUpdateTime == None` better to use `order.LastUpdateTime is None`

File: Algorithm.Python/UpdateOrderRegressionAlgorithm.py [hunk 2]
```diff
@@ -117,18 +117,19 @@ def OnData(self, data):
...
+            raise ValueError("Expected updated order LastUpdateTime to equal submitted update order event time");
```
Comment (jingwu74) [inline Algorithm.Python/UpdateOrderRegressionAlgorithm.py]: We can delete `;` in Python

File: Algorithm.Python/UpdateOrderRegressionAlgorithm.py [hunk 3]
```diff
@@ -117,18 +117,19 @@ def OnData(self, data):
 
     def OnOrderEvent(self, orderEvent):
         order = self.Transactions.GetOrderById(orderEvent.OrderId)
+        ticket = self.Transactions.GetOrderTicket(orderEvent.OrderId);
```
Comment (jingwu74) [inline Algorithm.Python/UpdateOrderRegressionAlgorithm.py]: We can delete `;` in Python

File: Algorithm.Python/UpdateOrderRegressionAlgorithm.py (hunk 2 above)
Comment (StefanoRaggi) [reply Algorithm.Python/UpdateOrderRegressionAlgorithm.py]: Whoops, copy/paste from C# :)

File: Algorithm.Python/CoarseFineFundamentalRegressionAlgorithm.py [hunk 4]
```diff
@@ -47,14 +47,15 @@ def Initialize(self):
         self.numberOfSymbolsFine = 2
 
     # return a list of three fixed symbol objects
-    def CoarseSelectionFunction(self, coarse):        
+    def CoarseSelectionFunction(self, coarse):
         tickers = [ "GOOG", "BAC", "SPY" ]
 
-        if self.Time < datetime(2014, 4, 5):
+        dt = datetime(self.Time.year, self.Time.month, self.Time.day)
+        if dt < datetime(2014, 4, 1):
```
Comment (AlexCatarino) [inline Algorithm.Python/CoarseFineFundamentalRegressionAlgorithm.py]: Alternatively you could use:
```python
from datetime import date
if self.Time.date() < date(2014, 4, 1):
```

**Pull request merged**